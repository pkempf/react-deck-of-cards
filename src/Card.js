import React, { useState } from "react";
import "./Card.css";

const Card = ({ id, name, imgSrc }) => {
  const [{ angle, xOffset, yOffset }] = useState({
    angle: Math.random() * 90 - 45,
    xOffset: Math.random() * 50 - 25,
    yOffset: Math.random() * 50 - 25,
  });

  const transform = `translate(${xOffset}px, ${yOffset}px) rotate(${angle}deg)`;
  return (
    <img
      className="Card"
      key={id}
      src={imgSrc}
      alt={name}
      style={{ transform }}
    />
  );
};

export default Card;
