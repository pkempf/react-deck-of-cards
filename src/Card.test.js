import { render } from "@testing-library/react";
import Card from "./Card";

beforeEach(() => {
  jest.spyOn(global.Math, "random").mockReturnValue(0.5);
});

afterEach(() => {
  jest.spyOn(global.Math, "random").mockRestore();
});

it("renders without crashing", () => {
  render(<Card />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<Card />);
  expect(asFragment()).toMatchSnapshot();
});
