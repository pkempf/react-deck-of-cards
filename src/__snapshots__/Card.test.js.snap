// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <img
    class="Card"
    style="transform: translate(0px, 0px) rotate(0deg);"
  />
</DocumentFragment>
`;
